swaybar/render.c

```diff
@@ -573,25 +621,28 @@ static uint32_t render_binding_mode_indicator(struct render_context *ctx,
 	cairo_set_operator(cairo, CAIRO_OPERATOR_SOURCE);
 	cairo_set_source_u32(cairo, config->colors.binding_mode.background);
 	ctx->background_color = config->colors.binding_mode.background;
-	cairo_rectangle(cairo, x, 0, width, height);
+	cairo_rectangle(cairo, *x, 0, width, height);
 	cairo_fill(cairo);
 
 	cairo_set_source_u32(cairo, config->colors.binding_mode.border);
-	cairo_rectangle(cairo, x, 0, width, border_width);
+	cairo_rectangle(cairo, *x, 0, width, border_width);
 	cairo_fill(cairo);
-	cairo_rectangle(cairo, x, 0, border_width, height);
+	cairo_rectangle(cairo, *x, 0, border_width, height);
 	cairo_fill(cairo);
-	cairo_rectangle(cairo, x + width - border_width, 0, border_width, height);
+	cairo_rectangle(cairo, *x + width - border_width, 0, border_width, height);
 	cairo_fill(cairo);
-	cairo_rectangle(cairo, x, height - border_width, width, border_width);
+	cairo_rectangle(cairo, *x, height - border_width, width, border_width);
 	cairo_fill(cairo);
 
 	double text_y = height / 2.0 - text_height / 2.0;
 	cairo_set_source_u32(cairo, config->colors.binding_mode.text);
-	cairo_move_to(cairo, x + width / 2 - text_width / 2, (int)floor(text_y));
+	cairo_move_to(cairo, *x + width / 2 - text_width / 2, (int)floor(text_y));
 	choose_text_aa_mode(ctx, config->colors.binding_mode.text);
 	pango_printf(cairo, config->font, output->scale,
 			output->bar->mode_pango_markup, "%s", mode);
+
+	*x += width;
```

Comment:
Does it really need to mutate `x`? Maybe you could just add width with later draw calls?

Reply:
I thought I keep it consistent with the other render functions.

Reply:
Yes, for example look at `render_status_line_text` or `render_workspace_button`

Reply:
Nevermind then